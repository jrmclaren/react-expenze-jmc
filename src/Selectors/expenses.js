import moment from 'moment';
// Get visible expenses
export default (expenses, {text, sortBy, startDate, endDate} ) => {
	return expenses.filter( ( expense ) => {
			const createdAtMoment = moment(expense.createdAt)
			let startDateMatch =  startDate? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
			let endDateMatch = endDate? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
			let textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
			// expenses.description as the text variable string inside of it
			// includes(); if one string includes another
			// convert both strings to lower case
			// 
			return startDateMatch && endDateMatch && textMatch;
	}).sort(( a, b  ) => {
		if( sortBy === 'date' ){
			return (a.createdAt < b.createdAt? 1 : -1);
		} else if( sortBy === 'amount') {
			return a.amount < b.amount? 1 : -1;
		}
	});
}
