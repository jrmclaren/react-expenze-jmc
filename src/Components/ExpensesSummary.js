import React from 'react';
import numeral from 'numeral';
import { selectExpensesTotal } from '../Selectors/expenses-total';
const ExpensesSummary = props => {
  let expenses = props.expenses;
  if (!expenses[0]) {
    return <p>No expenses</p>
  }
  return expenses.length == 1 ? (
    (<p>You are viewing 1 expense with a total of {numeral(expenses[0].amount / 100).format('$0,0.00')}</p>)
    ) : (
    (<p>{`You are viewing ${expenses.length} expenses with a total of ${numeral(selectExpensesTotal(expenses) / 100).format('$0,0.00')}`}</p>)
    )
}
export default ExpensesSummary;