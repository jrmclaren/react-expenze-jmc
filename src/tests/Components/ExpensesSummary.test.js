import React from 'react';
import ExpensesSummary from '../../Components/ExpensesSummary';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<ExpensesSummary expenses={expenses}/>);
});

test('should render default message correctly', () => {
  wrapper.setProps({
    expenses: {}
  });
  expect(wrapper).toMatchSnapshot();
});

test('should render single expense sumarry correctly', () => {
  let expense = [expenses[0]]
  wrapper.setProps({
    expenses: expense
  });
  expect(wrapper).toMatchSnapshot();
});

test('should render multiple expense summary correctly', () => {
  expect(wrapper).toMatchSnapshot();
});