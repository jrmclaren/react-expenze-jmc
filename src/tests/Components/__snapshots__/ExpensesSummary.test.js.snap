// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render default message correctly 1`] = `
<p>
  No expenses
</p>
`;

exports[`should render multiple expense summary correctly 1`] = `
<p>
  You are viewing 3 expenses with a total of $1,141.95
</p>
`;

exports[`should render single expense sumarry correctly 1`] = `
<p>
  You are viewing 1 expense with a total of 
  $1.95
</p>
`;
